src/pck/PckFile.h

```diff
@@ -98,6 +98,12 @@ class PckFile {
     std::optional<std::fstream> File;
     std::optional<std::ifstream> DataReader;
 
+    // PCK Format version number
+    // 0 = Godot 1.x, 2.x
+    // 1 = Godot 3.x
+    // 2 = Godot 4.x
+    uint32_t FormatVersion = 0;
```

Comment:
I think for when a new .pck file is created, this won't be read from a file causing this to stay at 0, which would create .pck files not loadable by Godot 3. I think initializing this variable to 1 would be the easiest option to handle that correctly.